changefeed: add and refine alerts (#15984)
pingcap/docs: tidb-cloud/monitor-built-in-alerting.md

|:--------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Changefeed latency is more than 600 seconds | Please check status of the Changefeed in ListPage and DetailPage of TiDB Cloud Console, there may be some error messages can help you diagnose what is happening. <br> There are several reasons can trigger this alarm. <br> 1. The overall traffic of upstream has increased, and the existing Changefeed specification is insufficient to handle it. If it is temporary traffic, it will automatically recover after a period of time. If it is continuous traffic, please scale up the Changefeed. <br> 2. Downstream and network abnormal, if it is this problem, please correct it first. <br> 3. Tables lack of indexes if downstream is RDS, which may cause low write performance and high latency, please add necessary indexes to upstream or downstream. <br> If the problem can not be fixed, please contact our support team directly for further assistance.   |

Reviewer: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed state in the ListPage and DetailPage of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alter include: <br> 1. The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br> 2. The downstream or network is abnormal. In this case, resolve this abnormality first. <br> 3. Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.   |
```

> | The changefeed latency exceeds 600 seconds. | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |

Reviewer: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed statuson the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |
```

> | The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for further assistance. |

Reviewer 2: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br/> If the problem cannot be fixed from your side, you can contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for further assistance. |
```

Reviewer 2: the list should be written in html tags `<ul><li></li></ul>`

> | The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br/> If the problem cannot be fixed from your side, you can contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for further assistance. |

Reviewer 2: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> Possible reasons that can trigger this alert include:<ul><li>The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed.</li><li>The downstream or network is abnormal. In this case, resolve this abnormality first.</li><li>Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream.</li></ul>If the problem cannot be fixed from your side, you can contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for further assistance. |
```

> | Changefeed status is failed                 | Please check status of the Changefeed in ListPage and DetailPage of TiDB Cloud Console, there may be some error messages can help you diagnose what is happening. <br> If the problem can not be fixed from your side, please contact our support team directly for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| The Changefeed state is `FAILED`.                 | Check the changefeed state in the ListPage and DetailPage of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

> | The changefeed status is `FAILED`.                 | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| The changefeed status is `FAILED`.                 | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

> | Changefeed status is warning                | Please check status of the Changefeed in ListPage and DetailPage of TiDB Cloud Console, there may be some error messages can help you diagnose what is happening. <br> If the problem can not be fixed from your side, please contact our support team directly for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| The changefeed state is  `WARNING`.              | Check the changefeed state in the ListPage and DetailPage of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```

> | The changefeed latency exceeds 600 seconds. | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |
> | The changefeed status is `FAILED`.                 | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
> | The changefeed status is  `WARNING`.              | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |
| The changefeed status is `FAILED`.                 | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| The changefeed status is  `WARNING`.              | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

> | The changefeed status is  `WARNING`.              | Check the changefeed status in the Changefeed Page and Changefeed Detail Page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| The changefeed status is  `WARNING`.              | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

> | The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |
> | The changefeed status is `FAILED`.                 | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
> | The changefeed status is `WARNING`.              | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| The changefeed latency exceeds 600 seconds. | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br> Possible reasons that can trigger this alert include: <br/> - The overall traffic in the upstream has increased, causing the existing changefeed specification to be insufficient to handle it. If the traffic increase is temporary, the changefeed latency will automatically recover after the traffic returns to normal. If the traffic increase is continuous, you need to scale up the changefeed. <br/> - The downstream or network is abnormal. In this case, resolve this abnormality first. <br/> - Tables lack indexes if the downstream is RDS, which might cause low write performance and high latency. In this case, you need to add the necessary indexes to the upstream or downstream. <br> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance. |
| The changefeed status is `FAILED`.                 | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact TiDB Cloud Support for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| The changefeed status is `WARNING`.              | Check the changefeed status on the **Changefeed** page and **Changefeed Detail** page of the TiDB Cloud console, where you can find some error messages to help diagnose this issue. <br/> If the problem cannot be fixed from your side, you can contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for further assistance.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```
